### PR TITLE
GPII-3846: Upgrade to couchdb 2.3

### PR DESCRIPTION
**Pull request by mrtyler**
Separated from https://github.com/gpii-ops/gpii-infra/pull/373.

### CODE REVIEW

**Comment by stepanstipl** [comment]
This PR still seems to contain a lot of `nginx-ingress` changes unrelated to CouchDB upgrade itself.

**Comment by amatas** [comment]
> This PR still seems to contain a lot of nginx-ingress changes unrelated to CouchDB upgrade itself.

This is because the upgrade of the nginx-ingress controller from upstream, right @mrtyler?

**Comment by mrtyler** [comment]
Per standup discussion, I have rebased this branch to remove extraneous commits. (Git will handle them fine during merge, but may have trouble if we need to revert.)

File: shared/versions.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ prometheus_to_sd:
 couchdb:
   upstream:
     repository: couchdb
-    tag: 2.2.0
+    tag: 2.3.0
```
Comment (stepanstipl) [inline shared/versions.yml]: Is there a reason not to use the latest 2.3.1?

Comment (mrtyler) [reply shared/versions.yml]: Not sure how I missed that version (release a couple weeks ago), but I'll stick with the version I've tested.

**Comment by amatas** [comment]
LGTM

**Comment by natarajaya** [comment]
LGTM!

**Pull request merged**